cmd/vault-staging-k8s-secret/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/teamsnap/vault-key/pkg/k8s"
+	"github.com/teamsnap/vault-key/pkg/vault"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// var env = map[string]map[string]string{}
+
+func main() {
+	cfg := zap.Config{
+		Level:            zap.NewAtomicLevelAt(zapcore.DebugLevel),
+		Encoding:         "console",
+		EncoderConfig:    zap.NewDevelopmentEncoderConfig(),
+		OutputPaths:      []string{"stderr"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+
+	logger, err := cfg.Build()
+	if err != nil {
+		panic(err)
+	}
+	cfg.Level.SetLevel(zap.ErrorLevel)
+
+	run(context.Background(), logger)
+}
+
+func run(ctx context.Context, lgr *zap.Logger) {
+	defaultEngine := os.Getenv("VAULT_SECRET")
+	lgr.Debug("vault-path to the default engine", zap.String("VAULT_SECRET", defaultEngine))
+	if len(defaultEngine) < 1 {
+		lgr.Fatal("You need to set VAULT_SECRET environment variable.")
+	}
+
+	override := os.Getenv("VAULT_SECRET_OVERRIDE")
+	lgr.Debug("vault-path to an overriding engine", zap.String("VAULT_SECRET_OVERRIDE", override))
+	if len(override) < 1 {
+		lgr.Info("No override engine provided.")
+	}
+
+	k8sNamespace := os.Getenv("K8S_NAMESPACE")
+	lgr.Debug("kubernetes namespace to apply secret", zap.String("K8S_NAMESPACE", k8sNamespace))
+	if len(k8sNamespace) < 1 {
+		lgr.Fatal("You need to set K8S_NAMESPACE environment variable.")
+	}
+
+	engines := []string{defaultEngine, override}
+	verifiedEngines := []string{}
+
+	for _, p := range engines {
+		lgr.Debug("verifying engine exists for path", zap.String("path", p))
+
+		if _, err := vault.ListEngines(ctx, p); err != nil {
+			verifiedEngines = append(verifiedEngines, p)
+		} else {
+			lgr.Error("cannot verify engine exists at path", zap.Error(err))
+		}
+	}
+
+	lgr.Info("getting secrets from vault from the verified engines", zap.Strings("verified-engines", verifiedEngines))
+	secrets := map[string]map[string]string{}
+	vault.GetSecrets(ctx, &secrets, verifiedEngines)
+
+	lgr.Info("creating overrides", zap.String("overrides", fmt.Sprintf("%v", secrets[override])))
+	mergedSecrets := secrets[override]
+
+	lgr.Info("setting defaults")
+	for k, v := range secrets[defaultEngine] {
+		if _, ok := mergedSecrets[k]; !ok {
+			mergedSecrets[k] = v
```

Comment:
We should add a note that the first secret wins out, or an affordance for reporting when conflicting secrets are detected and are being dropped.

Reply:
On L#70 I was dumping the entire override secret, but I like logging each one individually here better 👍 

Makes the output a little bit noisier but I think the verbosity when troubleshooting would be useful.